mono/utils/mono-compiler.h

```diff
@@ -160,7 +164,7 @@ typedef int32_t __mono_off32_t;
 #include <sys/mman.h>
 #endif
 
-#if !defined(mmap)
+#if !defined(mmap) && __NDK_MAJOR__ < 18
```

Comment:
1. Yes, because some versions of NDK don't declare it at all, some declare it for specific API levels and some (NDK r18) for all levels.
2. They declare it but it's an old trick to check #ifdef (which I think stops to work with some preprocessors though...)

Reply:
Declarations never produce a define. I have seen this ifdef check many times before, yes, but it never actually works. Unless, they also add an coincident define with the declaration. Declarations are not macros and vice versa.
```
void foo (...);
#define foo foo
```
then it works. With just regular `void foo (...)`, it is meaningless.

Reply:
It used to work universally, now it seems that it stopped to work with the newer versions of gcc and clang so, yes, it became meaningless.

Reply:
I've seen people write this pattern repeatedly, but it has never worked as intended.
Usually people are confused about the separation between C and the C preprocessor.
It "works", sure, but the ifdef is just always false, given only a declaration.
Unless a separate macro is thrown in, e.g. `#define mmap mmap64`.